internal/server/app.go

```diff
@@ -32,24 +29,34 @@ func Start() {
 		flag.PrintDefaults()
 		panic(err)
 	}
-	Storage = memstorage.NewMemStorage(ServerConfig.IsSyncDump(), ServerConfig.FileStoragePath)
-
-	// Восстанавливаем данные
-	if ServerConfig.RestoreStore {
-		Storage.Restore()
+	var storageClose func() error
+	if ServerConfig.DatabaseDSN == "" {
+		Storage, storageClose, err = memstorage.NewMemStorage(
+			ServerConfig.FileStoragePath,
+			ServerConfig.RestoreStore,
+			ServerConfig.StoreInterval,
+		)
+	} else {
+		// TODO: повторная инициализация при недоступности базы
+		Storage, storageClose, err = pgstorage.NewPGStorage(
+			ServerConfig.DatabaseDSN,
+		)
 	}
-	// Периодически сохраняем данные
-	if ServerConfig.DumpEnabled() && ServerConfig.StoreInterval > 0 {
-		go periodicDump()
+	if err != nil {
+		logger.Info("error creating storage:", err)
```

Comment:
```suggestion
		logger.Fatal("error creating storage:", err)
```
В этом случае можно смело и с грохотом падать.

Reply:
По заданию у сервера существует метод `/ping`, который работает при невозможности инициализации стораджа, поэтому просто упасть не можем.

По-хорошему, тут сделать повторные инициализации, но теста на это нет, в заданиях про это не говорится ничего, поэтому я ограничился TODO в комментарии чуть выше по коду.

Reply:
Неверное понимание задания. Метод ping нужен для мониторинга состояния соединения. Но если DSN указан неверно или сервер сразу недоступен, лучше не пытаться поднимать сервис (явно зависимый от БД). Потому как разрыв соединения - это диагностируемая (и устраняемая) проблема, а хендлер БД, инициализированный в nil - это поток паник в логах и необходимость все равно перезапустить сервер для инициализации соединения.

Reply:
@avhitrov  Указанная реализация  противоречит тесту десятого инкремента https://github.com/Yandex-Practicum/go-autotests/blob/main/cmd/metricstest/iteration10b_test.go